src/libraries/System.Net.Security/src/System/Net/Security/Pal.Android/SafeDeleteSslContext.cs

```diff
@@ -45,6 +52,12 @@ public SafeDeleteSslContext(SslAuthenticationOptions authOptions)
             try
             {
                 _sslContext = CreateSslContext(SslStreamProxy, authOptions);
+
+                // Create GC handle and store it in dictionary
+                GCHandle gcHandle = GCHandle.Alloc(this);
+                _nativeHandle = GCHandle.ToIntPtr(gcHandle);
+                s_contextMap[_nativeHandle] = gcHandle;
```

Comment:
This maps the GCHandle to itself. I do not see what it is trying to solve.

Reply:
Correct, the finalizer won't run if there is strong reference to the object. Should static dictionary contain WeakReference then?

Reply:
No GCHandle reference from managed side is needed.